build-system/common/utils.js

```diff
@@ -30,6 +33,31 @@ function logOnSameLine(message) {
   log(message);
 }
 
+/**
+ * Converts an array of globs to a list of matching files using gulp.src, which
+ * can handle negative globs.
+ *
+ * @param {!Array<string>} globs
+ * @return {!Array<string>}
+ */
+async function globsToFiles(globs) {
```

Comment:
What do you mean by "negative globs"? We use the `globs-to-files` npm package in other places in the code, does that not work here?

Reply:
> What do you mean by "negative globs"?

Something like `'!**/{node_modules,build,dist,dist.3p,dist.tools}/**'`.

> We use the `globs-to-files` npm package in other places in the code, does that not work here?

It doesn't. See "coming up" section in PR description.

Reply:
Note to self: also read PR descriptions 🙃

Reply:
Your comment was useful, in that it made me re-evaluate the wisdom of hacking together this implementation :)

A better and faster alternative that is being actively maintained is [`globby`](https://www.npmjs.com/package/globby). It supports everything we need, and can replace the function I wrote (and eventually, all the `globs-to-files` calls).

Reply:
Done in the latest commit.